.github/workflows/ci.yml

```diff
@@ -151,43 +152,105 @@ jobs:
     if: contains(needs.planner.outputs.jobs_to_run, 'kotlin')
     uses: ./.github/workflows/_kotlin.yml
     secrets: inherit
+
+  monitor-kotlin:
+    needs: [kotlin]
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - run: '[ "${{ needs.kotlin.result }}" = "failure" ] && gh run cancel ${{ github.run_id }}'
```

Comment:
There is no good way to de-duplicate this unfortunately that isn't actually more verbose. We could extract it into an action but then we'd have to pass the result in as a parameter which is actually two additional lines for each workflow.